### ISSUE TITLE
Does profiling work well with asyncio?

### ISSUE THREAD

**Issue by nchammas** [what-studio/profiling#26]
I noticed that the README mentions support for greenlets. How about Python 3.4+'s asyncio?

Should we just use `--timer=greenlet` for all the varieties of asynchronous programming that Python supports (greenlets, threading, multiprocessing, asyncio, etc.)?

**Comment by sublee** [comment]
No, you shouldn't use ``--timer=greenlet``.  I didn't care about ``asyncio`` application but I should have done it.  For now, probably we can't profile ``asyncio`` application with ``profiling``.  I'll try to find a solution.

Thanks to report.

**Comment by sublee** [comment]
I'm working on this issue.

Checkout https://github.com/sublee/profiling/tree/aware-of-eventloop and try like:

```console
$ profiling yourapp.py --eventloop-aware=asyncio
```

I need to hear some feedback.

**Comment by nchammas** [comment]
Hi @sublee! Thanks for revisiting this issue.

I won't be able to test this anytime soon, but I'll try to test it out at some point.

**Comment by thehesiod** [comment]
@sublee awesome!  Exactly what I was looking for.  Seems to work!  Thanks!

**Comment by Alexis-benoist** [comment]
@sublee I can't clone your version of the code doing `pip install  git+https://github.com/sublee/profiling/tree/eventloop-aware` doesn't clone the code.

    Collecting git+https://github.com/sublee/profiling/tree/eventloop-aware
      Cloning https://github.com/sublee/profiling/tree/eventloop-aware to    /var/folders/rs/75jmmsys1dl016fnw1gsq0r00000gn/T/pip-89loyzv3-build
    fatal: repository 'https://github.com/sublee/profiling/tree/eventloop-aware/' not found
      Complete output from command git clone -q https://github.com/sublee/profiling/tree/eventloop-aware /var/folders/rs/75jmmsys1dl016fnw1gsq0r00000gn/T/pip-89loyzv3-build:

**Comment by sublee** [comment]
@Alexis-benoist Your URL is not correct.  Try to install with `$ pip install git+https://github.com/what-studio/profiling.git@eventloop-aware`.

**Comment by ZetaHunter** [comment]
Any update on this? Could really use a profiling tool for asyncio.

**Comment by thehesiod** [comment]
I've been using http://www.pyvmmonitor.com/ with pretty good success

**Comment by zet4** [comment]
@thehesiod thanks for suggestion but I just tried it in my application and without any success.

**Comment by advance512** [comment]
+1 for this.

**Comment by lambdalisue** [comment]
Brilliant! Note that I needed to comment out `asyncio.Task._wakeup` and `asyncio.Task._step` in https://github.com/what-studio/profiling/blob/eventloop-aware/profiling/adapting.py#L30 to ignore `method_descriptor object has no attribute '__code__'` exception in Python 3.6

**Comment by xNinjaKittyx** [comment]
Hello, I ran into the same issue as above in Python 3.6. It would be nice to have this pulled into the code...

**Comment by MJ111** [comment]
@sublee 
https://github.com/sublee/profiling/tree/eventloop-aware this asyncio feature is merged to this repo master branch, right?
I'm confused cause I think it's working well now with asyncio.

**Comment by MJ111** [comment]
@sublee I think so. I thought it’s merged already because there’s asyncio.py in remote folder. I might be wrong.
